src/text/raw_text_reader.rs

```diff
@@ -480,170 +99,106 @@ impl<T: ToIonDataSource> IonReader for RawTextReader<T> {
     }
 
     fn next(&mut self) -> IonResult<RawStreamItem> {
-        // Parse the next value from the stream, storing it in `self.current_value`.
-        self.load_next_value()?;
-
-        // If we're positioned on an IVM, return the (major, minor) version tuple
-        if let Some((major, minor)) = self.current_ivm {
-            return Ok(RawStreamItem::VersionMarker(major, minor));
-        }
+        self.feed_if_needed()?;
+
+        match self.reader.next() {
+            Err(err @ IonError::IncompleteText { .. }) => {
+                // We received an incomplete.. and need more data..
+                loop {
+                    // If there is no more data to read, we need to bubble the incomplete up to the
+                    // caller.
+                    if 0 == self.read_source(1024)? {
```

Comment:
Why 1024 instead of whatever the buffer has room for?

Reply:
Arbitrarily chose 1024, but it isn't tied to the remaining space in the buffer in order to allow the buffer to grow to support reading blobs that are larger than the buffer.

Reply:
I wonder if we should double it on each retry so it ramps up faster for large values?